risc0/zkvm/src/host/server/prove/prover_impl.rs

```diff
@@ -96,31 +98,48 @@ impl ProverServer for ProverImpl {
             ))?
             .digest();
 
-        // Collect the proven assumption receipts from the Session.
-        // TODO(#982): Support unresolved assumptions here.
-        let assumption_receipts = session_assumption_receipts
+        let mut zkr_receipts = HashMap::new();
+        for proof_request in session.pending_zkrs.iter() {
+            let receipt = prove_zkr(&proof_request.control_id, &proof_request.input)?;
+            let claim_digest = receipt.claim.digest();
+            zkr_receipts.insert(claim_digest, receipt);
```

Comment:
Using only the claim digest as the key might not distinguish two incompatible receipts. In particular, if two versions of the same accelerator are used, or if there simply happens to be a collision in the claim format between two ZKRs (not something we can prevent in general, as it's easy to create a second ZKR that matches the claim structure of another), then this will break down. Using the `Assumption` (i.e. the `(control_root, claim_digest)` tuple) here would be one way to avoid this. The `control_id` could also be instead of the `control_root` to create the key.